wasm/store.go

```diff
@@ -789,13 +789,19 @@ func (s *Store) buildExportInstances(module *Module, target *ModuleInstance) (ro
 
 // ValidateAddrRange checks if the given address range is a valid address range.
 // It accepts rangeSize as uint64 so that callers can add or multiply two uint32 addresses
-// without overflow. eg.) ValidateAddrRange(uint32Offset, uint64(uint32Size) + 1)
+// without overflow. For example, `ValidateAddrRange(uint32Offset, uint64(uint32Size) + 1)`.
+// ValidateAddrRange works even if `m` is nil so that memory range validation
+// against a module with no memory exported can be done in a consistent way.
 func (m *MemoryInstance) ValidateAddrRange(addr uint32, rangeSize uint64) bool {
+	if m == nil {
```

Comment:
would this be less awkward if we instead made sure this is never nil and use a constant `MemoryInstance(0,0,[]byte{})` instead?

Reply:
Starting with the memory length zero but later growing is valid so nil (memory is not declared in module) and memory with length zero is different

Reply:
But yeah max zero sounds ok for non memory case

Reply:
yeah because I noticed earlier it is valid to decode max zero, so if we have a bug where are writing we'll need to fix it anyway

Reply:
yeah, the idea of always having at least  0-max memory sounds good. I'll change the code.

Reply:
considering that @codefromthecrypt is doing refactoring on `MemoryInstance`, how about @codefromthecrypt doing the introduction of `NoMemoryInstance` constant at the same time and close this PR? It will conflict in a lot of places.

Reply:
you mean #204? works for me

Reply:
only thing is that my brain is on another change, so probably won't do 204 until tomorrow or weds

Reply:
yes, I mean #204. It sounds ok to me. Thanks!